node/core/dispute-coordinator/src/db/v1.rs

```diff
@@ -258,20 +323,116 @@ pub(crate) fn note_current_session(
 	Ok(())
 }
 
+/// Until what session votes have been cleaned up already.
+///
+/// That is the db has already been purged of votes for sessions older than the returned
+/// `SessionIndex`.
+fn load_cleaned_votes_watermark(
+	db: &dyn Database,
+	config: &ColumnConfiguration,
+) -> FatalResult<Option<SessionIndex>> {
+	load_decode(db, config.col_data, CLEANED_VOTES_WATERMARK_KEY)
+		.map_err(|e| FatalError::DbReadFailed(e))
```

Comment:
why introduce a separate error type? other methods use
`SubsystemError::with_origin("dispute-coordinator", e)`

we already have `SubsystemError::IoError` and `Error::Io`, no reason to introduce yet another copy of that error

Reply:
I needed it compatible with the `FatalResult` used in `write`. The errors here are a mixture of legacy and "new" error types, this should get cleaned up at some point indeed.